rust/scx_utils/examples/mangolog.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+
+// This software may be used and distributed according to the terms of the
+// GNU General Public License version 2.
+use scx_utils::mangoapp::mangoapp_msg_v1;
+
+use anyhow::bail;
+use anyhow::Result;
+use libc::{msgget, msgrcv, IPC_NOWAIT};
+
+use std::mem;
+use std::path::Path;
+use std::thread;
+use std::time::Duration;
+use std::{ffi::CString, io};
+
+fn main() -> Result<()> {
+    // NOTE: the mangoapp file has to be present when gamescope launches in order to read from the
```

Comment:
nice trick, although I only tested this with mangohud enabled. IDK if that data is emitted from gamescope when the overlay is off, that might be worth testing idk.

Reply:
It does seem to work without `mangohud` with
```
gamescope -e -W 1920 -H 1080 -r 60 --mangoapp -- %command%
```
with `mangohud` via:
```
gamescope -e -W 1920 -H 1080 -r 60 --mangoapp -- mangohud %command%
```
It was super annoying to figure that out

Reply:
Ahh when I say mangohud I mean mangoapp/the gamescope kinda-sorta vendored version of mangohud. I didn't intentionally test w/o that, but in my C version of this, things would hang when I ran this on machines w/o gamescope accidentally. 

That being said, you mentioned this is example code in which case, that doesn't matter so much.